tests/test_trailedEdgeSources.py

```diff
@@ -0,0 +1,220 @@
+        self.bbox = lsst.geom.Box2I(lsst.geom.Point2I(-20, -30),
+                                    lsst.geom.Extent2I(140, 160))
+        self.dataset = TrailedTestDataset(self.bbox)
+
+        # Trail which extends into edge pixels
+        self.trail = TrailedEdgeSource(100000.0, self.length, self.theta, self.xc, self.yc)
+        self.dataset.addTrailedSource(self.trail)
+
+    def tearDown(self):
+        del self.center
+        del self.bbox
+        del self.trail
+        del self.dataset
+
+    def makeTrailedSourceMeasurementTask(self, plugin=None, dependencies=(),
+                                         config=None, schema=None,
+                                         algMetadata=None):
+        """Set up a measurement task for a trailed source plugin.
+        """
+
+        config = self.makeSingleFrameMeasurementConfig(plugin=plugin,
+                                                       dependencies=dependencies)
+
+        # Make sure the shape slot is base_SdssShape
+        config.slots.shape = "base_SdssShape"
+        return self.makeSingleFrameMeasurementTask(plugin=plugin,
+                                                   dependencies=dependencies,
+                                                   config=config,
+                                                   schema=schema,
+                                                   algMetadata=algMetadata)
+
+    def testEdgeFlag(self):
+        """Test the NaivePlugin with one trailed source crossing edge pixels
+        and one source which does not.
+
+        Given a `TrailedTestDataset`, run the NaivePlugin measurement and
+        check that the trailed sources have the edge flag set.
+        """
+
+        # Set up and run Naive measurement.
+        task = self.makeTrailedSourceMeasurementTask(
+            plugin="ext_trailedSources_Naive",
+            dependencies=("base_SdssCentroid", "base_SdssShape")
+        )
+        exposure, catalog = self.dataset.realize(5.0, task.schema,
+                                                 randomSeed=0)
+        task.run(catalog, exposure)
+        record = catalog[0]
+
+        # Check that x0, y0 or x1, y1 is flagged as an edge pixel
+        x1 = np.round(record['ext_trailedSources_Naive_x1'])
+        y1 = np.round(record['ext_trailedSources_Naive_y1'])
+        x0 = np.round(record['ext_trailedSources_Naive_x0'])
+        y0 = np.round(record['ext_trailedSources_Naive_y0'])
+
+        edge_pixel_set = np.array([(exposure.mask[x0, y0] & exposure.mask.getPlaneBitMask(
+            'EDGE') != 0), (exposure.mask[x1, y1] & exposure.mask.getPlaneBitMask('EDGE') != 0)])
+
+        # Checks trailed which crosses into edge pixels
+        if record['truth_x'] == 100:
+
+            self.assertTrue(edge_pixel_set.any())
+
+            # Make sure measurement edge flag is set, but Naive_flag is not.
+            # A failed trailed source measurement can still have the edge flag
```

Comment:
What would having the new edge flag set mean in the situation of a failed measurement? Would it be worthwhile to add a null test, i.e., one where a measurement fails and the flag is not errantly set as a result of the failure? Part of this question is coming from an uncertainty of what failed measurement type situations would ever cause `Naive_flag` to be set...